### PR TITLE
Example using Array.prototype.fill() on empty array

### LINKED ISSUE #7371: Issue with "Array.prototype.fill()": (short summary here please)
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill

#### What information was incorrect, unhelpful, or incomplete?

Description of the `end` parameter was incomplete, leading to extra debugging time.

#### Specific section or headline?

Description

#### What did you expect to see?

(in addition to existing description)

The `end` parameter is treated as `array.length` if it is greater than `array.length` - you can't lengthen an array using the `fill` method without setting `array.length` to the required length first.

#### Did you test this? If so, how?

Yes.  By evalaluating

``` js
let hash = [];
hash.length = 256;  // omit and the array is not modified
console.log(hash.fill(0, 0, 256) );
```

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/array/fill`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/array/fill/index.md
* Last commit: https://github.com/mdn/content/commit/cf607d68522cd35ee7670782d3ee3a361eaef2e4
* Document last modified: 2021-07-20T18:47:57.000Z

</details>

### PR DESCRIPTION
**Pull request by Anurella**
> Issue number to be fixed (ie 'Fixes #123', if there is an associated issue)
Fixes #7371

> What was wrong/why is this fix needed? (quick summary only)
Added a note and example to highlight the use of Array.prototype.fill() on an empty array.

> Anything else that could help us review it

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -217,8 +217,8 @@ and use it to make a pull request.
 
 1. Next, you'll want to start the local preview service, so you can see
 the changes you'll make as they would look in production. Once started,
-this local preview service is available at `http://localhost:5000`
-within your browser.
+this local preview service is available at `http://localhost:5000/`
```
Comment (escattone) [inline README.md:220]: 👍

File: README.md [hunk 2]
```diff
@@ -217,8 +217,8 @@ and use it to make a pull request.
...
+within your browser. Do not forget to run `npm install` for dependencies.
```
Comment (escattone) [inline README.md:221]: Installing the dependencies is already covered by the code below, and we use `yarn` for that.
```suggestion
within your browser.
```

File: .gitignore [hunk 3]
```diff
@@ -6,3 +6,6 @@ yarn-debug.log*
 yarn-error.log*
 _githistory.json
 .DS_Store
+changes.md
+package-lock.json
```
Comment (Ryuno-Ki) [inline .gitignore:10]: How this came to be created?
Are you using `yarn` or `npm` locally?

File: files/en-us/web/api/element/queryselector/index.html [hunk 4]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
 <pre class="brush:js">var el = document.body.querySelector("style[type='text/css'], style:not([type])");
 </pre>
 
+
+
+<h3 id="Get_direct_descendants_using_scope_pseudo_class"> Get direct descendants using :scope pseudo-class</h3>
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:78]:
```suggestion
<h3 id="Get_direct_descendants_using_scope_pseudo_class">Get direct descendants using :scope pseudo-class</h3>
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 5]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<p> This example uses the <a href="/en-US/docs/Web/CSS/:scope">:scope</a> pseudo-class to
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:79]:
```suggestion
<p>This example uses the <a href="/en-US/docs/Web/CSS/:scope">:scope</a> pseudo-class to 
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 6]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+      &lt;span&gt;Love is Selfless.&lt;/span&gt;
+    &lt;/span&gt;
+  &lt;/div&gt;
+&lt;/div&gt;
+</pre>
+
+<h4 id="CSS">CSS</h4>
+<pre class="brush: css">
+  span {
+    display:block;
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:100]:
```suggestion
    display: block;
```

File: files/en-us/web/javascript/reference/global_objects/array/fill/index.md [hunk 7]
```diff
@@ -48,6 +48,11 @@ The modified array, filled with `value`.
 - `fill` is a mutator method: it will change the array itself and return it, not a copy of it.
 - If the first parameter is an object, each slot in the array will reference that object.
 
+<div class="note notecard" markdown="1">
```
Comment (Ryuno-Ki) [inline files/en-us/web/javascript/reference/global_objects/array/fill/index.md:51]: The `markdown="1"` looks foreign here.
I can't recall having seen that anywhere else.
But I'm not used to Markdown files here yet, either, so take it with a grain of salt.

File: files/en-us/web/javascript/reference/global_objects/array/fill/index.md [hunk 8]
```diff
@@ -48,6 +48,11 @@ The modified array, filled with `value`.
...
+Note: Using Array.Fill() on an empty array would not modify it as the array has nothing to be modified. To use Array.Fill() when declaring an array, make sure to assign slots to the array.
```
Comment (Ryuno-Ki) [inline files/en-us/web/javascript/reference/global_objects/array/fill/index.md:52]:
```suggestion
Note: Using Array.fill() on an empty array would not modify it as the array has nothing to be modified. To use Array.fill() when declaring an array, make sure to assign slots to the array.
```

Comment (Ryuno-Ki) [reply files/en-us/web/javascript/reference/global_objects/array/fill/index.md:52]: Or perhaps even `Array.prototype.fill()`

**Review by Ryuno-Ki — DISMISSED** [review]
This will definitively need another round.
Please make sure you're working with `yarn`: https://github.com/mdn/content#more-substantial-changes (item 6) or ask for assistance.

File: .gitignore (hunk 3 above)
Comment (escattone) [inline .gitignore:10]: Please remove these. I think these are better placed in your personal `.git/info/exclude` file.

Comment (Anurella) [reply .gitignore:10]: using yarn, this has been removed locally. will make another push

**Review by escattone — CHANGES_REQUESTED** [review]
@Anurella My review only covers the `.gitignore`, `README.md`, and `package-lock.json` files. I'll leave the content review to @Rumyra.

We use `yarn` rather than `npm`, so please remove the `package-lock.json` file from this PR.

File: files/en-us/web/javascript/reference/global_objects/array/fill/index.md (hunk 7 above)
Comment (Anurella) [reply files/en-us/web/javascript/reference/global_objects/array/fill/index.md:51]: The file is in complete markdown, I wanted to create a note ( there is an existing style with the class note ). I added the markdown=1 to ensure I could still use markdown in the containing div, would remove it.

File: files/en-us/web/javascript/reference/global_objects/array/fill/index.md (hunk 8 above)
Comment (Anurella) [reply files/en-us/web/javascript/reference/global_objects/array/fill/index.md:52]: will make the correction

File: .gitignore (hunk 3 above)
Comment (Anurella) [reply .gitignore:10]: thank you will make the corrections

Comment (escattone) [reply .gitignore]: Thanks @Anurella!

**Review by escattone — APPROVED** [review]

**Comment by Rumyra** [comment]
I think there are some files here which shouldn't be: The querySelector index and the CSS scope page. @Anurella it's possible two of your fixes have ended up in this PR (I noticed one fix for the `:scope` bits else where). Would you be able to remove those changes from this PR?

Thanks!

**Comment by Anurella** [comment]
> I think there are some files here which shouldn't be: The querySelector index and the CSS scope page. @Anurella it's possible two of your fixes have ended up in this PR (I noticed one fix for the `:scope` bits else where). Would you be able to remove those changes from this PR?
> 
> Thanks!

I am not sure how to do that, please how do I do that ?

**Comment by Rumyra** [comment]
@Anurella  😬 I have done this once before (you're not alone) and honestly I can't remember how I untangled things 😆

If those changes have been merged, if you pull the latest main from mdn/content and merge into this branch, I'm _pretty_ sure those files won't show up as changed. If not ping me here and I'll look with fresh eyes tomorrow 👍

**Comment by Ryuno-Ki** [comment]
FWIW we are adding some more lines to the paragraph to outline some strategies on rebasing and merge conflicting in context of PRs at the moment.

Hopefully, this will lead to less issues from contributors.

**Review by teoli2003 — APPROVED** [review]

**Comment by teoli2003** [comment]
I fixed the trivial merge conflicts and landed this.

**Comment by soffieprincess07** [comment]
> This will definitively need another round.
> Please make sure you're working with `yarn`: https://github.com/mdn/content#more-substantial-changes (item 6) or ask for assistance.

**Comment by pswan004** [comment]
> > Issue number to be fixed (ie 'Fixes #123', if there is an associated issue)
> > Fixes #7371
> 
> > What was wrong/why is this fix needed? (quick summary only)
> > Added a note and example to highlight the use of Array.prototype.fill() on an empty array.
> 
> > Anything else that could help us review it

**Pull request merged**